### ISSUE TITLE
Print unit cell volume and its standard deviation at end of processing

### ISSUE THREAD

**Issue by benjaminhwilliams** [xia2/xia2#442]
A request from the CX gang at DLS, especially for the case of in-beam structural changes.  Does anyone object to my including the unit cell volume in the processing summary?  Should I limit this to small molecule mode?

**Comment by Anthchirp** [comment]
If you are poking around in that area: with the move to Python 3 xia2 started to sometimes show UC angles as "90.0000000000000000000" degrees.

**Comment by benjaminhwilliams** [comment]
Yes, I investigated exactly that on Tuesday afternoon.  A little investigation revealed the problem to be slightly more complicated than the Python 3 `__str__` == `__repr__` that I was expecting.

**Comment by rjgildea** [comment]
The answer to the excessive precision in unit cell angles lies somewhere in this function:
https://github.com/cctbx/cctbx_project/blob/26feebb2605a6a5932261c81959f6831c8a665e8/libtbx/utils.py#L1787-L1822
Probably some clues into the Python 2 vs 3 differences here:
https://stackoverflow.com/questions/25898733/why-does-strfloat-return-more-digits-in-python-3-than-python-2

**Comment by Anthchirp** [comment]
quite possibly. Behaviour if `standard_uncertainty=0` is to `return str(value)`, and

Python 2:
```python
>>> str(90.0+1e-14)
'90.0'
```
Python 3:
```python
>>> str(90.0+1e-14)
'90.00000000000001'
```

Suggest copying function into `dials.util` and fixing it up there

**Comment by benjaminhwilliams** [comment]
Yep, so much I discovered on Tuesday.  A possible fig leaf would be to set the default precision when `standard_uncertainty=0` to the same as Python 2's `__str__` for floats (1 x 10⁻¹²-ish? I can't remember).  Do we prefer that to actually applying the relevant constraints to `dxtbx.model.crystal`'s calculated unit cell?

**Comment by benjaminhwilliams** [comment]
Basically, there are two points here, firstly we obviously don't care about physically meaningless precision so we might as well hide it, but secondly we might not need to hide it if we applied the relevant constraints in the first place.

**Comment by graeme-winter** [comment]
@Anthchirp thanks for the `testset150` example set - annoyingly it failed to exhibit the issue:

```
Assuming spacegroup: P 1 2 1
Unit cell (with estimated std devs):
32.248(8)  55.793(12) 38.762(7)
90.0      108.23(3)   90.0     
mtz_unmerged format:
Scaled reflections (NATIVE): /Users/graeme/Projects/excessive-precision/work0/DataFiles/AUTOMATIC_DEFAULT_scaled_unmerged.mtz
mtz format:
Scaled reflections: /Users/graeme/Projects/excessive-precision/work0/DataFiles/AUTOMATIC_DEFAULT_free.mtz
Processing took 00h 01m 24s
XIA2 used... ccp4, dials, dials.scale, xia2
Here are the appropriate citations (BIBTeX in xia2-citations.bib.)
Beilsten-Edmands, J. et al. (2020) Acta Cryst. D76.
Winn, M. D. et al. (2011) Acta Cryst. D67, 235-242.
Winter, G. (2010) J. Appl. Cryst. 43, 186-190.
Winter, G. et al. (2018) Acta Cryst. D74, 85-97.
Status: normal termination
```

Additional info:

```
Grey-Area work0 :) $ dials.python
Python 3.6.10 | packaged by conda-forge | (default, Apr  6 2020, 14:40:13) 
[GCC Clang 9.0.1 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 
```

**Comment by graeme-winter** [comment]
However `dials.show` on the `12_integrated.expt` manifests the error rather nicely 

```
Crystal:
    Unit cell: 32.242(6), 55.746(5), 38.775(7), 89.99999999999999, 108.243(10), 89.99999999999999
    Space group: P 1 2 1
    U matrix:  {{-0.5542, -0.8058, -0.2087},
                {-0.7141,  0.3315,  0.6166},
                {-0.4276,  0.4908, -0.7591}}
    B matrix:  {{ 0.0310,  0.0000,  0.0000},
                {-0.0000,  0.0179,  0.0000},
                { 0.0102, -0.0000,  0.0272}}
    A = UB:    {{-0.0193, -0.0145, -0.0057},
                {-0.0158,  0.0059,  0.0167},
                {-0.0210,  0.0088, -0.0206}}
    A sampled at 151 scan points
  Average unit cell: (32.2355, 55.7423, 38.7678, 90, 108.255, 90)
Profile model:
    type: gaussian_rs
    delta_b (sigma_b): 0.071422 (0.023807)
    delta_m (sigma_m): 0.464993 (0.154998)
```

**Comment by benjaminhwilliams** [comment]
Does `dials.show` use the same `libtbx.utils.format_float_with_standard_uncertainty`?

**Comment by graeme-winter** [comment]
Indirectly, yes, via `dxtbx.model`

**Comment by benjaminhwilliams** [comment]
The original question was unrelated to precision of printed floats.  @graeme-winter, do you have an opinion on my including the unit cell volume & s.d. in the summary statistics?

**Comment by graeme-winter** [comment]
Unit cell volume seems harmless as an addition - small molecule people want it, so I don't see why others may object (except those who seek with scripts which run xia2) -> if you make sure there is a news fragment is fine by me, or we could make it a "small_molecule=True" thing?

**Comment by Anthchirp** [comment]
Volume could be useful for MX as well - even though most people will know how to run Matthew's it might be helpful for an eyeball estimate?
